### PR TITLE
TF-2671 signature clipped when composing long content

### PR DESCRIPTION
**Pull request by tddang-linagora**
### Issue
- #2671 

## Dependent

- Need merged: https://github.com/linagora/enough_html_editor/pull/29

### Demo
**iOS**

https://github.com/linagora/tmail-flutter/assets/160106668/33e955dd-99b6-4f4a-b13e-cb5cae98be96

https://github.com/linagora/tmail-flutter/assets/160106668/1c730350-7fb0-4b49-889d-9e82bdd1523d

https://github.com/linagora/tmail-flutter/assets/160106668/88598358-ff54-47b3-b8f1-6b8aacf2de78

**Android**

https://github.com/linagora/tmail-flutter/assets/160106668/cf9e28e7-210b-4f5f-9989-254548115bfd

### CODE REVIEW

**Comment by dab246** [comment]
Please also test on iOS

**Comment by tddang-linagora** [comment]
> Please also test on iOS

The demo is iOS simulator

**Comment by dab246** [comment]
> > Please also test on iOS
> 
> The demo is iOS simulator

Oh, so Android?

**Comment by tddang-linagora** [comment]
> > > Please also test on iOS
> > 
> > 
> > The demo is iOS simulator
> 
> Oh, What about Android?

Updated

**Comment by dab246** [comment]
- [ ] For case `Reply, Forward` with long email content.

File: lib/features/composer/presentation/composer_controller.dart [hunk 1]
```diff
@@ -1648,8 +1648,13 @@ class ComposerController extends BaseController with DragDropFileMixin {
...
     BuildContext context,
   ) {
+    final scrollTarget = realCoordinateY -
+      (responsiveUtils.isLandscapeMobile(context)
+        ? 0
+        : headerEditorMobileHeight / 2);
+    final maxScrollExtend = scrollController.position.maxScrollExtent;
     scrollController.jumpTo(
-      realCoordinateY - (responsiveUtils.isLandscapeMobile(context) ? 0 : headerEditorMobileHeight / 2),
+      scrollTarget > maxScrollExtend ? maxScrollExtend : scrollTarget,
```
Comment (dab246) [inline lib/features/composer/presentation/composer_controller.dart:1657]:
```suggestion
Math.min(scrollTarget, maxScrollExtend)
```

File: lib/features/composer/presentation/composer_view.dart [hunk 2]
```diff
@@ -210,7 +210,7 @@ class ComposerView extends GetWidget<ComposerController> {
                               return const SizedBox.shrink();
                             }
                           }),
-                          const SizedBox(height: ComposerStyle.keyboardMaxHeight),
+                          SizedBox(height: MediaQuery.viewInsetsOf(context).bottom),
```
Comment (dab246) [inline lib/features/composer/presentation/composer_view.dart:213]: Also remove keyboardMaxHeight in ComposerStyle if not used

**Comment by tddang-linagora** [comment]
> * [x]  For case `Reply, Forward` with long email content.

Added demos

File: lib/features/composer/presentation/composer_controller.dart (hunk 1 above)
Comment (tddang-linagora) [reply lib/features/composer/presentation/composer_controller.dart]: Done

File: lib/features/composer/presentation/composer_view.dart (hunk 2 above)
Comment (tddang-linagora) [reply lib/features/composer/presentation/composer_view.dart:213]: Done

File: pubspec.yaml [hunk 3]
```diff
@@ -55,7 +55,7 @@ dependencies:
   rich_text_composer:
     git:
       url: https://github.com/linagora/rich-text-composer.git
-      ref: cnb_supported
+      ref: enhancement/update_enough_html_editor_dependency
```
Comment (dab246) [inline pubspec.yaml:58]: Don't need change it. Only update `pubspec.lock`
```
flutter pub upgrade rich_text_composer
```

**Review by hoangdat — COMMENTED** [review]
- [ ] please update pubspec

**Review by dab246 — APPROVED** [review]

**Pull request merged**